tests/system/large/operations/test_semantics.py

```diff
@@ -447,8 +596,40 @@ def test_join(instruction, session, gemini_flash_model):
     )
 
 
+@pytest.mark.parametrize(
+    ("reply"),
+    [
+        pytest.param("y"),
+        pytest.param(
+            "n", marks=pytest.mark.xfail(raises=exceptions.OperationAbortedError)
+        ),
+    ],
+)
+def test_join_with_confirmation(session, gemini_flash_model, reply, monkeypatch):
+    bigframes.options.experiments.semantic_operators = True
+    bigframes.options.compute.semantic_ops_confirmation_threshold = 0
```

Comment:
Because our tests are multi-threaded, this has the possibility of introducing flakiness to our test suite.

Please use a `with bpd.option_context(...)` block to set these values. That will ensure thread-local values are used.

See: https://github.com/googleapis/python-bigquery-dataframes/blob/10f08daec6034aafe48096be56683c953accc79a/tests/system/small/test_progress_bar.py#L34-L35

Reply:
Thanks for the example. Code updated